mlir/include/mlir/Dialect/EmitC/Transforms/Passes.td

```diff
@@ -21,4 +21,15 @@ def FormExpressions : Pass<"form-expressions"> {
   let dependentDialects = ["emitc::EmitCDialect"];
 }
 
+def EliminateLibm : Pass<"eliminate-libm"> {
```

Comment:
```suggestion
def EliminateLibm : Pass<"eliminate-libm", "ModuleOp"> {
```
to ensure that this pass is only scheduled on ModuleOps.
Then you can also remove the check in the pass itself.